sdks/jreleaser-java-sdk-commons/src/test/java/org/jreleaser/sdk/commons/AbstractMavenDeployerTest.java

```diff
@@ -0,0 +1,197 @@
+            assertTrue(cause.getMessage().contains(expectedMessage), "Actual message: " + cause.getMessage());
+        } catch (Exception e) {
+            fail(e);
+        }
+    }
+    private JReleaserContext context;
+
+    @BeforeEach
+    void setup() {
+        context = mock(JReleaserContext.class, RETURNS_DEEP_STUBS);
+        when(context.getLogger()).thenReturn(mock(JReleaserLogger.class));
+    }
+
+
+    static class TestMavenDeployer extends AbstractMavenDeployer<org.jreleaser.model.api.deploy.maven.MavenDeployer, org.jreleaser.model.internal.deploy.maven.MavenDeployer<org.jreleaser.model.api.deploy.maven.MavenDeployer>> {
+        private final org.jreleaser.model.internal.deploy.maven.MavenDeployer<org.jreleaser.model.api.deploy.maven.MavenDeployer> mavenDeployer;
+
+        TestMavenDeployer(JReleaserContext context, org.jreleaser.model.internal.deploy.maven.MavenDeployer<org.jreleaser.model.api.deploy.maven.MavenDeployer> mavenDeployer) {
+            super(context);
+            this.mavenDeployer = mavenDeployer;
+        }
+
+        @Override
+        public org.jreleaser.model.internal.deploy.maven.MavenDeployer<org.jreleaser.model.api.deploy.maven.MavenDeployer> getDeployer() {
+            return mavenDeployer;
+        }
+
+        @Override
+        public String getType() {
+            return "test";
+        }
+
+        public void setDeployer(org.jreleaser.model.internal.deploy.maven.MavenDeployer<org.jreleaser.model.api.deploy.maven.MavenDeployer> deployer) {}
+
+        @Override
+        public void deploy(String s) {}
+    }
+
+    @Test
+    void throwsExceptionWhenKeyExpired() {
+        try (MockedStatic<SigningUtils> signingUtils = mockStatic(SigningUtils.class)) {
+            signingUtils.when(() -> SigningUtils.getPublicKeyID(any())).thenReturn(Optional.of("ABCDEF"));
+            signingUtils.when(() -> SigningUtils.getFingerprint(any())).thenReturn(Optional.of("ABCDEF123456"));
+            // Set expiration date in the past
+            signingUtils.when(() -> SigningUtils.getExpirationDateOfPublicKey(any()))
+                .thenReturn(Optional.of(Instant.now().minusSeconds(3600)));
+
+
+            org.jreleaser.model.internal.deploy.maven.MavenDeployer<org.jreleaser.model.api.deploy.maven.MavenDeployer> mavenDeployer =
+                mock(org.jreleaser.model.internal.deploy.maven.MavenDeployer.class);
+            TestMavenDeployer deployer = new TestMavenDeployer(context, mavenDeployer);
+
+            Throwable thrown = assertThrows(Exception.class, () -> {
+                var method = AbstractMavenDeployer.class.getDeclaredMethod("verifyKeyIsValid");
+                method.setAccessible(true);
+                method.invoke(deployer);
+            });
+            Throwable cause = thrown;
+            if (cause instanceof java.lang.reflect.InvocationTargetException && cause.getCause() != null) {
```

Comment:
This seems arbitrary. Better use a loop checking if the cause is not an `InvocationTargetException`